File: 3442. Maximum Difference Between Even and Odd Frequency I/maxDifference.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
    public:
        int maxDifference(string s) {
            vector<int> arr(26,0);
            int maxOdd = 0;
            int minEven = 101;
    
            for(auto chr : s){
                arr[chr-'a']++;
            }
    
            for(auto num : arr){
                if(num % 2 == 0 && num > 0){
                    minEven = min(minEven, num);
                }
                else{
                    maxOdd = max(maxOdd, num);
                }
            }
    
            return maxOdd - minEven;
        }
    };

File: 3442. Maximum Difference Between Even and Odd Frequency I/maxDifference.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDifference = function (s) {
    let arr = new Array(26).fill(0)

    for (let chr of s) {
        let code = chr.charCodeAt(0) - 97
        arr[code]++
    }
    //odd - even
    let maxOdd = 0
    let minEven = Infinity

    for (let num of arr) {
        if (num % 2 === 0 && num > 0) {
            minEven = Math.min(minEven, num)
        }
        else {
            maxOdd = Math.max(maxOdd, num)
        }
    }

    return maxOdd - minEven
};